src/util/metrics/jemalloc_metrics.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Mutex;
+
+use prometheus::core::{Collector, Desc};
+use prometheus::proto::MetricFamily;
+use prometheus::{IntGaugeVec, Opts, Result};
+
+use tikv_alloc;
+
+pub fn monitor_jemalloc_stats<S: Into<String>>(namespace: S) -> Result<()> {
+    prometheus::register(Box::new(JemallocStatsCollector::new(namespace)?))
+}
+
+struct JemallocStatsCollector {
+    descs: Vec<Desc>,
+    metrics: Mutex<IntGaugeVec>,
+}
+
+impl JemallocStatsCollector {
+    fn new<S: Into<String>>(namespace: S) -> Result<JemallocStatsCollector> {
+        let ns = namespace.into();
+        let stats = IntGaugeVec::new(
+            Opts::new("jemalloc_stats", "Jemalloc stats").namespace(ns.clone()),
+            &["type"],
+        )?;
+        Ok(JemallocStatsCollector {
+            descs: stats.desc().into_iter().cloned().collect(),
+            metrics: Mutex::new(stats),
+        })
+    }
+}
+
+impl Collector for JemallocStatsCollector {
+    fn desc(&self) -> Vec<&Desc> {
+        self.descs.iter().collect()
+    }
+
+    fn collect(&self) -> Vec<MetricFamily> {
```

Comment:
How frequently this function will be called?

Reply:
Grafana is showing a data point every 30 seconds, but I don't really know where it is configured. @overvenus do you happen to know?

Reply:
@zhangjinpeng1987 It is collected as the same frequency of our remaining metrics.